src/Creator.tsx

```diff
@@ -0,0 +1,244 @@
+        name: 'test-quickstart',
+        ...(selectedTypeMeta?.extraMetadata ?? {}),
+      },
+      spec: {
+        ...baseQuickStart.spec,
+        tasks: tasks,
+      },
+    },
+    { taskErrors },
+  ];
+}
+
+const Creator = () => {
+  const [rawType, setRawType] = useState<ItemKind | null>(null);
+
+  const [rawQuickStart, setRawQuickStart] = useState<QuickStart>({
+    metadata: {
+      name: 'test-quickstart',
+    },
+    spec: {
+      displayName: '',
+      icon: null,
+      description: '',
+    },
+  });
+
+  const selectedType =
+    rawType !== null ? { id: rawType, meta: itemKindMeta[rawType] } : null;
+
+  const [bundles, setBundles] = useState<string[]>([]);
+  const [taskContents, setTaskContents] = useState<string[]>([]);
+
+  const [currentTask, setCurrentTask] = useState<number | null>(null);
+
+  const updateSpec = (
+    updater: (old: QuickStartSpec) => Partial<QuickStartSpec>
+  ) => {
+    setRawQuickStart((old) => ({
+      ...old,
+      spec: {
+        ...old.spec,
+        ...updater(old.spec),
+      },
+    }));
+  };
+
+  const setType = (newType: ItemKind | null) => {
+    if (newType !== null) {
+      const meta = itemKindMeta[newType];
+
+      setRawQuickStart((old) => {
+        const updates: Partial<QuickStart> = {};
+
+        updates.spec = { ...old.spec };
+
+        updates.spec.type = {
+          text: meta.displayName,
+          color: meta.tagColor,
+        };
+
+        if (
+          meta.hasTasks &&
+          (old.spec.tasks === undefined || old.spec.tasks.length === 0)
+        ) {
+          updates.spec.tasks = [EMPTY_TASK];
+        }
+
+        if (!meta.hasTasks) {
+          updates.spec.tasks = undefined;
+          updates.spec.introduction = undefined;
+          updates.spec.prerequisites = [];
+        }
+
+        if (!meta.fields.url) updates.spec.link = undefined;
+        if (!meta.fields.duration) updates.spec.durationMinutes = undefined;
+
+        updates.metadata = { ...BASE_METADATA, ...meta.extraMetadata };
+
+        return { ...old, ...updates };
+      });
+
+      if (meta.hasTasks) {
+        setTaskContents((old) => (old.length === 0 ? [''] : old));
+      } else if (!meta.hasTasks) {
+        setTaskContents([]);
+      }
+    }
+
+    setRawType(newType);
+  };
+
+  const [quickStart, errors] = useMemo(
+    () => makeDemoQuickStart(rawType, rawQuickStart, taskContents),
+    [rawType, rawQuickStart, taskContents]
+  );
+
+  const files = useMemo(() => {
+    const effectiveName = quickStart.spec.displayName
+      .toLowerCase()
+      .replaceAll(/\s/g, '-')
+      .replaceAll(/(^-+)|(-+$)/g, '');
+
+    const adjustedQuickstart = { ...quickStart };
+    adjustedQuickstart.spec = { ...adjustedQuickstart.spec };
+    adjustedQuickstart.metadata = {
+      ...adjustedQuickstart.metadata,
+      name: effectiveName,
+    };
+
+    delete adjustedQuickstart.spec['icon'];
+
+    return [
+      {
+        name: 'metadata.yaml',
```

Comment:
There is an extra `-` in the metadata file showing in the code component as well as in the downloaded file:
![screenshot-stage foo redhat com_1337-2024 07 10-09_26_38](https://github.com/RedHatInsights/learning-resources/assets/22619452/60543904-81c1-47dd-896f-06077acf39a1)

```yaml
kind: QuickStarts
name: as
tags:
  - - kind: bundle
      value: openshift
```